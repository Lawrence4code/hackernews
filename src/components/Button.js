import React from 'react';

const Button = ({ className = '', onClick, children }) => (
  <button onClick={onClick} type="type" className={className}>
    {children}
  </button>
);

export default Button;
