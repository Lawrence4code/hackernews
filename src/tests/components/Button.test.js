import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../../components/Button';

Enzyme.configure({ adapter: new Adapter() });

describe('Button Component', () => {
  test('should render Button component without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('should render Button correctly', () => {
    const component = renderer.create(<Button> Awesome Button </Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have button element', () => {
    const wrapper = shallow(<Button className="test" />);
    expect(wrapper.find('.test').length).toBe(1);
  });
});
