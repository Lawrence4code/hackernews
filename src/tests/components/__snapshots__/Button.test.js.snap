// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render Button correctly 1`] = `
<button
  className=""
  type="button"
>
   Awesome Button 
</button>
`;
